### PR TITLE
Gpl 806 add all negative plates in add to dart migration

### LINKED ISSUE #202: GPL-806 Update Add to DART Migration
**User story**
The migration to add legacy samples to DART should meet the requirements slightly differently. This will ensure plates are added to DART that don't have any positive samples, so the DART data set is as complete as possible. It should:

* Get all samples between the specified timestamp (not just Result=Positive ones)
* Remove cherrypicked samples from this list of samples (already implemented)
* ... (already implemented)
* Group samples by plate (already implemented)
* Add the plate to DART (already implemented)
* Add only Result=Positive samples to DART

**Who are the primary contacts for this story**
Chris F
Andrew S

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] Add to DART sample migration should be updated to implement above steps

**Dependencies**
No dependencies

**Additional context**
No additional context

### PR DESCRIPTION
**Pull request by Chris-Friend**
Closes #202 

Changes proposed in this pull request:

* Ensure we add all-negative plates to DART during the add-to-DART migration
* As a result, migration is also updated to add UUID fields to all samples, not just Result=Positive ones. Is this an issue? I don't think it is

### CODE REVIEW

**Comment by pjvv** [comment]
> As a result, migration is also updated to add UUID fields to all samples, not just Result=Positive ones. Is this an issue? I don't think it is

I don't think so either; just some extra data.

File: migrations/helpers/update_filtered_positives_helper.py [hunk 1]
```diff
@@ -243,17 +243,18 @@ def update_dart_fields(config: ModuleType, samples: List[Sample]) -> bool:
...
-                            well_props = map_mongo_doc_to_dart_well_props(sample)
-                            set_dart_well_properties(
-                                cursor, plate_barcode, well_props, well_index  # type:ignore
-                            )
-                        else:
-                            raise ValueError(
-                                "Unable to determine DART well index for sample "
-                                f"{sample[FIELD_ROOT_SAMPLE_ID]} in plate {plate_barcode}"
-                            )
+                        if sample[FIELD_RESULT] == POSITIVE_RESULT_VALUE:
```
Comment (pjvv) [inline migrations/helpers/update_filtered_positives_helper.py:246]: Remind me why we don't just check for filtered positive here?

Comment (Chris-Friend) [reply migrations/helpers/update_filtered_positives_helper.py:246]: It stems from updating the filtered positive rules: if an unpicked sample is changed from fit to pick to not fit to pick, we wouldn't end up updating that sample in DART, as it wouldn't be filtered positive

Comment (pjvv) [reply migrations/helpers/update_filtered_positives_helper.py:246]: Oh yes, that rings a bell, thank you Chris.

**Review by pjvv — APPROVED** [review]

**Review by andrewsparkes — COMMENTED** [review]
Bit late but looks good to me too!

File: migrations/helpers/dart_samples_update_helper.py [hunk 2]
```diff
@@ -150,16 +151,13 @@ def migrate_all_dbs(config, s_start_datetime: str = "", s_end_datetime: str = ""
         logger.exception(e)
 
 
-def get_positive_samples(
-    samples_collection: Collection, start_datetime: datetime, end_datetime: datetime
-) -> List[Sample]:
+def get_samples(samples_collection: Collection, start_datetime: datetime, end_datetime: datetime) -> List[Sample]:
     logger.debug(f"Selecting positive samples between {start_datetime} and {end_datetime}")
```
Comment (KatyTaylor) [inline migrations/helpers/dart_samples_update_helper.py:155]: Think this debug message should have the word 'positive' removed

Comment (pjvv) [reply migrations/helpers/dart_samples_update_helper.py:155]: True.

Comment (Chris-Friend) [reply migrations/helpers/dart_samples_update_helper.py:155]: Will address this in a new, small PR

**Pull request merged**